### PR TITLE
Beginning C++23: Chapter 11

### PR DESCRIPTION
**Pull request by Peter-VanWeert**

### CODE REVIEW

File: Examples/Modules/Chapter 11/Ex11_06/Ex11_06.cpp [hunk 1]
```diff
@@ -1,18 +1,14 @@
...
 }
 
 int main()
 {
-  std::cout << "math::sqrt2 has the value " << math::sqrt2 << std::endl;
-  std::cout << "This should be 0: " << (math::sqrt2 - std::numbers::sqrt2) << std::endl;
-  std::cout << "This should be 2: " << math::square(math::sqrt2) << std::endl;
+  std::println("math::sqrt2 has the value {}", math::sqrt2);
+  std::println("This should be 0: {}", math::sqrt2 - std::numbers::sqrt2);
 }
```
Comment (MarcGregoire) [inline Examples/Modules/Chapter 11/Ex11_06/Ex11_06.cpp:14]: The text is still saying something about the removed call to math::square()

File: Examples/Modules/Chapter 11/Ex11_06A/Ex11_06A.cpp [hunk 2]
```diff
@@ -1,11 +1,9 @@
...
 import squaring;
 
 int main()
 {
-  std::cout << "math::sqrt2 has the value " << math::sqrt2 << std::endl;
-  std::cout << "This should be 0: " << (math::sqrt2 - std::numbers::sqrt2) << std::endl;
-  std::cout << "This should be 2: " << math::square(math::sqrt2) << std::endl;
+  std::println("math::sqrt2 has the value {}", math::sqrt2);
+  std::println("This should be 0: {}", math::sqrt2 - std::numbers::sqrt2);
 }
```
Comment (MarcGregoire) [inline Examples/Modules/Chapter 11/Ex11_06A/Ex11_06A.cpp:9]: Add back the call to math::square()?

File: Examples/Modules/Chapter 11/Ex11_06B/Ex11_06B.cpp [hunk 3]
```diff
@@ -1,11 +1,9 @@
...
 import squaring;
 
 int main()
 {
-  std::cout << "math::sqrt2 has the value " << math::sqrt2 << std::endl;
-  std::cout << "This should be 0: " << (math::sqrt2 - std::numbers::sqrt2) << std::endl;
-  std::cout << "This should be 2: " << math::square(math::sqrt2) << std::endl;
+  std::println("math::sqrt2 has the value {}", math::sqrt2);
+  std::println("This should be 0: {}", math::sqrt2 - std::numbers::sqrt2);
 }
```
Comment (MarcGregoire) [inline Examples/Modules/Chapter 11/Ex11_06B/Ex11_06B.cpp:9]: Add back the call to math::square()?

File: Examples/Modules/Chapter 11/Ex11_08/Ex11_08.cpp [hunk 4]
```diff
@@ -17,8 +17,7 @@ auto hypotenuse(const auto& x, const auto& y)
 
 int main()
 {
-  std::cout << "math::sqrt2 has the value " << math::sqrt2 << std::endl;
-  std::cout << "This should be 0: " << (math::sqrt2 - std::numbers::sqrt2) << std::endl;
-  std::cout << "This should be 2: " << math::square(math::sqrt2) << std::endl;
-  std::cout << "This should be 5: " << hypotenuse(3, 4) << std::endl;
+  std::println("math::sqrt2 has the value {}", math::sqrt2);
+  std::println("This should be 0: {}", math::sqrt2 - std::numbers::sqrt2);
+  std::println("This should be 5: {}", hypotenuse(3, 4));
```
Comment (MarcGregoire) [inline Examples/Modules/Chapter 11/Ex11_08/Ex11_08.cpp:22]: Add back the call to math::square()?

**Review by MarcGregoire — CHANGES_REQUESTED** [review]
Done with review.

File: Exercises/Modules/Chapter 11/Soln11_06/words.cppm [hunk 5]
```diff
@@ -1,16 +1,14 @@
...
 {
   using Words = std::vector<std::shared_ptr<std::string>>;
 
   void sort(Words& words);
   void extract_words(Words& words, const std::string& text, const std::string& separators);
-  void show_words(const Words& words);
+  void print_words(const Words& words);
 }
 
 export namespace w = words;
```
Comment (Peter-VanWeert) [inline Exercises/Modules/Chapter 11/Soln11_06/words.cppm:14]: The `export using namespace words;` below should actually work...

File: Examples/Modules/Chapter 11/Ex11_03/Ex11_03.cpp [hunk 6]
```diff
@@ -1,27 +1,30 @@
...
+  // Even though the std::string and std::string_view names are not visible,
+  // their reachability implies that the names of their constructors and destructors (see Chapter 12) are visible...
+  from_roman("DCLXVI"); // Creates std::string_view object to invoke from_roman(string_view)
+  to_roman(1234);      // Receives and destructs the string object returned by to_roman(int)
 
+  // In fact, the name of any member function, not just the constructor or destructor,
+  // of a reachable type is visible. The name of the size() member of std::string is therefore visible,
+  // meaning this member function can therefore be invoked...
+  [[maybe_unused]]
+  const std::size_t num_numerals = to_roman(1234).size();  // 8 numerals ("MCCXXXIV")
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Examples/Modules/Chapter 11/Ex11_03/Ex11_03.cpp:18]: Should be auto. std::size_t is not visible... technically...

File: Examples/Modules/Chapter 11/Ex11_03/Ex11_03.cpp [hunk 7]
```diff
@@ -1,27 +1,30 @@
...
+  
+  // Even though you cannot use the invisible std::string / std::string_view names,
+  // you can still capture objects of these types using auto / const auto&...
+  
+// std::string_view s{ "MMXX" };   /* Error: the name std::string_view is not visible */
+// std::string roman{ to_roman(567) };  /* Error: the name std::string is not visible */
+  
   auto roman{ to_roman(567) };
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Examples/Modules/Chapter 11/Ex11_03/Ex11_03.cpp:26]: Now both say 789...

File: Examples/Modules/Chapter 11/Ex11_03/Ex11_03.cpp [hunk 8]
```diff
@@ -1,27 +1,30 @@
...
-  std::cout << "567 in Roman numerals is " << roman.c_str() << std::endl;
-
-  // std::cout << "std::stoi() is not visible: " << std::stoi("1234") << std::endl;
-
-  // The << operator (which is implemented as a non-member function) is not visible either
-  // std::cout << "1234 in Roman numerals is " << to_roman(1234) << std::endl;
+  auto first_X = roman.find('X');  // 2 (index of first X in "DLXVII")
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Examples/Modules/Chapter 11/Ex11_03/Ex11_03.cpp:27]: "first X" also made no sense anymore. Hence the switch to 789.

**Pull request merged**